pkg/sql/sem/builtins/builtins.go

```diff
@@ -1243,6 +1245,90 @@ var builtins = map[string]builtinDefinition{
 		"Calculates the CRC-32 hash using the Castagnoli polynomial.",
 	),
 
+	"digest": makeBuiltin(
+		tree.FunctionProperties{Category: categoryCrypto},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"data", types.String}, {"type", types.String}},
+			ReturnType: tree.FixedReturnType(types.Bytes),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				alg := tree.MustBeDString(args[1])
+				hashFunc, err := getHashFunc(string(alg))
+				if err != nil {
+					return nil, err
+				}
+				h := hashFunc()
+				if ok, err := feedHash(h, args[:1]); !ok || err != nil {
+					return tree.DNull, err
+				}
+				return tree.NewDBytes(tree.DBytes(h.Sum(nil))), nil
+			},
+			Info: "Computes a binary hash of the given `data`. `type` is the algorithm " +
+				"to use (md5, sha1, sha224, sha256, sha384, or sha512).",
+			Volatility: tree.VolatilityLeakProof,
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"data", types.Bytes}, {"type", types.String}},
+			ReturnType: tree.FixedReturnType(types.Bytes),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				alg := tree.MustBeDString(args[1])
+				hashFunc, err := getHashFunc(string(alg))
+				if err != nil {
+					return nil, err
+				}
+				h := hashFunc()
+				if ok, err := feedHash(h, args[:1]); !ok || err != nil {
+					return tree.DNull, err
+				}
+				return tree.NewDBytes(tree.DBytes(h.Sum(nil))), nil
+			},
+			Info: "Computes a binary hash of the given `data`. `type` is the algorithm " +
+				"to use (md5, sha1, sha224, sha256, sha384, or sha512).",
+			Volatility: tree.VolatilityLeakProof,
+		},
+	),
+
+	"hmac": makeBuiltin(
+		tree.FunctionProperties{Category: categoryCrypto},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"data", types.String}, {"key", types.String}, {"type", types.String}},
+			ReturnType: tree.FixedReturnType(types.Bytes),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				key := tree.MustBeDString(args[1])
+				alg := tree.MustBeDString(args[2])
+				hashFunc, err := getHashFunc(string(alg))
+				if err != nil {
+					return nil, err
+				}
+				h := hmac.New(hashFunc, []byte(key))
+				if ok, err := feedHash(h, args[:1]); !ok || err != nil {
+					return tree.DNull, err
+				}
+				return tree.NewDBytes(tree.DBytes(h.Sum(nil))), nil
+			},
+			Info:       "Calculates hashed MAC for `data` with key `key`. `type` is the same as in `digest()`.",
+			Volatility: tree.VolatilityLeakProof,
+		},
+		tree.Overload{
+			Types:      tree.ArgTypes{{"data", types.Bytes}, {"key", types.Bytes}, {"type", types.String}},
+			ReturnType: tree.FixedReturnType(types.Bytes),
+			Fn: func(_ *tree.EvalContext, args tree.Datums) (tree.Datum, error) {
+				key := tree.MustBeDBytes(args[1])
+				alg := tree.MustBeDString(args[2])
+				hashFunc, err := getHashFunc(string(alg))
```

Comment:
i think we should move this into a `pkg/util/crypto` package so that they can be reused. but it's not too important atm; i'll leave it up to you.

Reply:
i think this would be good to do if we actually implement more crypto functions.